spot_rl_experiments/spot_rl/utils/heuristic_nav.py

```diff
@@ -0,0 +1,458 @@
+        vision_T_hand: mn.Matrix4,
+        body_T_hand: mn.Matrix4,
+        cam_intrinsics,
+        spot: Spot,
+        angle: float = 0.0,
+    ):
+
+        rgb_img_vis = rgb_img.copy() if self.visualize else None
+
+        det_status, det = self.object_detection(rgb_img, object_target, angle)
+        if not det_status:
+            return False, (None, None, None), rgb_img_vis
+        x1, y1, x2, y2, conf = det
+        x1_int, y1_int, x2_int, y2_int = int(x1), int(y1), int(x2), int(y2)
+
+        (u, v), z = get_best_uvz_from_detection(unscaled_depth, [x1, y1, x2, y2])
+
+        if self.visualize:
+            rgb_img_vis = cv2.rectangle(
+                rgb_img_vis,
+                (x1_int, y1_int),
+                (x2_int, y2_int),
+                color=(0, 0, 255),
+                thickness=2,
+            )
+
+        if z >= 0.5 and z <= 2.5:
+            binary_depth_img = np.where(hand_depth_img > 0, 1, 0)
+            binary_depth_img = np.uint8(binary_depth_img)
+            mixed_image = rgb_img * binary_depth_img[:, :, None]
+            point_in_local_3d = get_3d_point(cam_intrinsics, (u, v), z)
+            (
+                corner_det_status,
+                best_z,
+                best_corner_yx,
+                other_best_yxs,
+            ) = get_z_offset_by_corner_detection(
+                mixed_image, unscaled_depth, [x1, y1, x2, y2], z
+            )
+            offset = point_in_local_3d[-1] - best_z if corner_det_status else 0
+            point_in_local_3d[-1] -= offset + self.corner_static_offset
+            (global_x, global_y), theta = convert_point_from_local_to_global_nav_target(
+                point_in_local_3d, None, spot, vision_T_hand, body_T_hand
+            )
+            if self.visualize:
+                if corner_det_status:
+                    rgb_img_vis = cv2.circle(
+                        rgb_img_vis,
+                        (best_corner_yx[-1], best_corner_yx[0]),
+                        7,
+                        (0, 0, 255),
+                        thickness=-1,
+                    )
+                    for yx in other_best_yxs:
+                        rgb_img_vis = cv2.circle(
+                            rgb_img_vis, (yx[-1], yx[0]), 1, (255, 0, 0)
+                        )
+                org = (0, y1_int - 50)
+                text = "({:.2f}, {:.2f}, {:.2f}), O:{:.2f}, Z:{:.2f}".format(
+                    global_x, global_y, np.degrees(theta), offset, z
+                )
+                rgb_img_vis = cv2.putText(
+                    rgb_img_vis,
+                    text=text,
+                    org=org,
+                    fontFace=cv2.FONT_HERSHEY_SIMPLEX,
+                    fontScale=0.9,
+                    color=(0, 0, 255),
+                    thickness=2,
+                )
+            return True, (global_x, global_y, theta), rgb_img_vis
+
+        return False, (None, None, None), rgb_img_vis
+
+
+def navigate_to_aria_goal(
```

Comment:
where do we call `navigate_to_aria_goal`?

Reply:
Sorry I forgot to remove that function, this was shifted to spotskillmanager.py